python/main.py

```diff
@@ -258,6 +272,189 @@ def insert_item(item: Item):
+    items = cursor.fetchall()
+
+    db_gen.close()
+
+
+    # print(items)
+    # items = [(1, 'jacket', 'fashion', 'ad55d25f2c10c56522147b214aeed7ad13319808d7ce999787ac8c239b24f71d.jpg'), 
+    # (2, 'pen', 'stationery', 'ad55d25f2c10c56522147b214aeed7ad13319808d7ce999787ac8c239b24f71d.jpg'), 
+    # (3, 'bottoms', 'fashion', 'ad55d25f2c10c56522147b214aeed7ad13319808d7ce999787ac8c239b24f71d.jpg')]
+    
+    lst = []
+    for i in items:
+        if len(i) == 4:
+            ele = GetItemimage(name=i[1],category = i[2],image_name= i[3])
+        else:
+            ele = GetItemimage(name=i[1],category = i[2],image_name= "Nan")
+        lst.append(ele)
+
+    resjson = GetimageItemResponse(items = lst)
+
+    return resjson
+"""
+
+"""
+# change post end point for 5-1
+@app.post("/items",response_model = AddItemResponse)
+async def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
+
+    # print("kakuninn!!!",image)
+    file_name = image.filename
+    # Create image path
+    image_path = images / file_name
+
+
+    if not image.filename.endswith(".jpg"):
+        raise HTTPException(status_code=400, detail="Image path does not end with .jpg")
+
+    if not image_path.exists():
+        logger.info(f"Image not found: {image}")
+        image = images / "default.jpg"
+
+    # hashlibの引数はバイナリを読み込むので一度データを読み込む必要がある
+    file_hash = hashlib.sha256()
+    contents = await image.read()  # ファイルの内容を読み込む
+    file_hash.update(contents)
+    file_hash_name= file_hash.hexdigest()+".jpg"
+
+    # connect database
+    db_gen = get_db()  # get_db()を呼び出してジェネレータを取得
+    conn = next(db_gen)  # ジェネレータを消費してデータベース接続を取得
+    cursor = conn.cursor()
+
+    print("kakuninnshite!!!!")
+    # insert to database
+"""
+    # cursor.execute("""INSERT INTO items (name, category, image_name) VALUES (?, ?, ?)""", (name, category, file_hash_name))
+"""
+    # saved change data
+    conn.commit()
+    db_gen.close()
+    
+    return AddItemResponse(**{"message": f"item received: {name,category,file_name}"})
+"""
+
+
+# add search end point for 5-2
+@app.get("/search",response_model = GetimageItemResponse)
+def search_items(keyword:str):
+
+    # connect database
+    db_gen = get_db()  # get_db()を呼び出してジェネレータを取得
+    conn = next(db_gen)  # ジェネレータを消費してデータベース接続を取得
+    cursor = conn.cursor()
+
+    # fetch item table from database
+    cursor.execute("SELECT * FROM items WHERE name = ?",(keyword,))
+    items = cursor.fetchall()
+
+    db_gen.close()
+    
+    lst = []
+    for i in items:
+        if len(i) == 4:
+            ele = GetItemimage(name=i[1],category = i[2],image_name= i[3])
+        else:
+            ele = GetItemimage(name=i[1],category = i[2],image_name= "Nan")
+        lst.append(ele)
+
+    resjson = GetimageItemResponse(items = lst)
+
+    return resjson
+
+# changed get endpoint for 5-3
+@app.get("/items",response_model = GetimageItemResponse)
+def get_items(db: sqlite3.Connection = Depends(get_db))-> List[dict]:
```

Comment:
この部分の型注釈が間違っているので、削除するか正しい型注釈に置き換えてください。

```suggestion
def get_items(db: sqlite3.Connection = Depends(get_db)):
```